### ISSUE TITLE
False positive: Access to generic instance variable through class is ambiguous

### ISSUE THREAD

**Issue by thomasLeMeur** [microsoft/pyright#8083]
**Describe the bug**
Since v1.1.348, false-positives can occur  on `Access to generic instance variable through class is ambiguous` error with inheritance.

This bug report follows https://github.com/microsoft/pyright/issues/7089 issue and this [comment](https://github.com/microsoft/pyright/issues/7089#issuecomment-2017034030) 

**Code or Screenshots**
```python
from typing import Generic, TypeVar

T = TypeVar("T")

class Base(Generic[T]):      some_attr: T
class StringBase(Base[str]): some_attr = "foo"
class IntBase(Base[int]):    some_attr = 0

assert StringBase.some_attr == "foo" # `Access to generic instance variable through class is ambiguous`
assert IntBase.some_attr    == 0     # `Access to generic instance variable through class is ambiguous`
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release of pyright.

**Comment by thomasLeMeur** [comment]
Thank you, my Lord !

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.367.
